lib/models/finamp_models.dart

```diff
@@ -1721,3 +1721,171 @@ enum TranscodeDownloadsSetting {
   @HiveField(2)
   ask;
 }
+
+
+/// A reference to a downloadable lyric with no state.  Can be freely created
+/// from a BaseItemDto and LyricDto at any time.  LyricsStubs/LyricsItems are considered
+/// equivalent if their types and ids match.
+@JsonSerializable(
```

Comment:
You are correct that this is far more complicated than needed.  I think the below code should be all you really need.  The caller should already have the isarId to use from the DownloadStub, so just require it in the constructor. You might even be able to directly embed the LyricDto with the embedded tag and avoid the overhead of JSON.  I had problems embedding BaseItemDtos due to the limitations on embedded objects in Isar, but LyricDtos are much simpler and might work.
```
@collection
class DownloadedLyrics {
 DownloadedLyrics._build({
    required this.jsonItem,
    required this.isarId,
  });

  factory DownloadedLyrics.fromItem({
    required int isarId,
    required LyricDto item,
  }) {
    return DownloadedLyrics._build(
        isarId: isarId),
        jsonItem: jsonEncode(item.toJson()),
    );
  }

  /// The integer ID used as a database key by Isar
  final Id isarId;

  /// The LyricDto as a JSON string for storage in isar.
  /// Use [lyricDto] to retrieve.
  final String? jsonItem;

  @ignore
  LyricDto? get lyricDto => _lyricDtoCached ??=
      ((jsonItem == null) ? null : LyricDto.fromJson(jsonDecode(jsonItem!)));
  @ignore
  LyricDto? _lyricDtoCached;
}
```